### PR TITLE
Rewrite project to TypeScript

### PR DESCRIPTION
**Pull request by prichodko**

### CODE REVIEW

**Comment by felicio** [comment]
PR as is should provide a snippet under `psr` prefix? I can't see it in the menu.

**Comment by prichodko** [comment]
Right now it has no snippets, so it couldn't provide any.

File: src/extension.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  provideCompletionItems(): ProviderResult<CompletionItem[]> {
+    return this.snippets.map(s => new PSCompletionItem(s))
+  }
+}
+
+export async function activate(context: ExtensionContext) {
+  let registerCompletionProvider = languages.registerCompletionItemProvider(
+    'javascript',
+    new PSCompletionProvider(snippets as [Snippet]),
```
Comment (felicio) [inline src/extension.ts]: Shouldn't this at least register completion items and display (empty) snippets under 'psr' prefix?

Comment (felicio) [inline src/extension.ts:44]: Shouldn't this at least register completion items and display (empty) snippets under 'psr' prefix?

File: src/extension.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  body: string
+  description: string
+}
+
+let registerCompletionProvider: Disposable | undefined
+
+class PSCompletionItem extends CompletionItem {
+  constructor(snippet: Snippet) {
+    super('psr', CompletionItemKind.Snippet)
+  }
```
Comment (felicio) [inline src/extension.ts]: I tried to add a snippet body here, but without success. What else has to be done for the sample snippet from `snippets.json` to be registered?

File: src/extension.ts (hunk 1 above)
Comment (felicio) [reply src/extension.ts:44]: I see, you're not sending an actual array yet. To see it in action I replaced `snippets` for `[snippets.consoleLog]`.

File: src/extension.ts (hunk 2 above)
Comment (felicio) [inline src/extension.ts]: `insertedText` and `detail` properties are left out intentionally?

**Comment by prichodko** [comment]
Good catch, I forgot that `snippets.json` is not of `Array` type. Try now.

**Review by felicio — APPROVED** [review]

**Pull request merged**